### DISCUSSION TITLE
SSH identity file destroyed with default settings

### DISCUSSION THREAD

**Discussion by tronkko** [deployphp/deployer#3978] [Bugs] [answered]
### Deployer Version

v7.5.8

### Target OS

Arch Linux

### Which PHP version are you using?

PHP 8.3

### Content of deploy.php or deploy.yaml

```
<?php

namespace Deployer;

host('staging')
    ->set('hostname', 'hops');

task('whoami', function () {
    run('whoami');
});
```

### Steps to reproduce

1. Create an SSH identity file named hops

        cd ~/.ssh
        ssh-keygen -f hops

3. Ensure that hops and hops.pub exist the .ssh directory
4. Cd to the project directory containing the above deploy.php file
5. Run deployer as

        ./vendor/deployer/deployer/bin/dep whoami

After this step the SSH identity file ~/.ssh/hops is gone preventing you from connecting to the host again!

The problem is fixed by specifying the ssh control path in the deploy.php file as

    host('staging')
        ->set('hostname', 'hops')
        ->set('ssh_control_path', '~/.ssh/hops.socket');

I hope that you change the default control path to avoid overwriting identity files by accident.  For example, add ".socket" suffix to the file name or change the directory ~/.ssh/deployer or similar.

**Comment by antonmedv** [comment, marked as answer]
This sounds reasonable. Will add a fix.
